R/truncate_linelist.R

```diff
@@ -59,71 +43,84 @@
 #' linelist <- sim_linelist()
 #' linelist_trunc <- truncate_linelist(linelist)
 #'
-#' # set maximum date to apply truncation to 2023-01-01
-#' linelist_trunc <- truncate_linelist(linelist, max_date = "2023-01-01")
-#'
-#' # apply longer truncation to hospital admission
+#' # set truncation point 3 weeks before the end of outbreak
 #' linelist_trunc <- truncate_linelist(
 #'   linelist,
-#'   delay = function(x) rlnorm(n = x, meanlog = 2, sdlog = 0.5),
-#'   truncation_event = "admission"
+#'   truncation_time = 3,
+#'   unit = "weeks"
 #' )
 #'
-#' # variable right truncation with mean 2 and sd 1 (default behaviour)
+#' # set truncation point to 2 months since the start of outbreak
 #' linelist_trunc <- truncate_linelist(
 #'   linelist,
-#'   delay = function(x) rlnorm(n = x, meanlog = 0.58, sdlog = 0.47)
+#'   truncation_time = 2,
+#'   unit = "months",
+#'   direction = "forwards"
 #' )
 #'
-#' # fixed right truncation of 5 days
+#' # set truncation point to 2023-03-01
 #' linelist_trunc <- truncate_linelist(
 #'   linelist,
-#'   delay = function(x) rep(5, n = x)
+#'   truncation_time = as.Date("2023-03-01")
 #' )
 truncate_linelist <- function(linelist,
-                              delay = function(x) stats::rlnorm(n = x, meanlog = 0.58, sdlog = 0.47), # nolint line_length_linter
-                              max_date = NULL,
-                              truncation_event = c("reporting", "onset",
-                                                   "admission", "outcome")) {
+                              truncation_time = 14,
+                              unit = c("days", "weeks", "months", "years"),
+                              direction = c("backwards", "forwards")) {
+  arg_ignore <- missing(unit) && missing(direction)
   .check_linelist(linelist)
-  .check_func_req_args(func = delay, func_name = "delay", n_req_args = 1)
-  truncation_event <- match.arg(truncation_event)
-  if (is.null(max_date)) {
-    date_cols <- grep(pattern = "date_", x = colnames(linelist), fixed = TRUE)
-    max_date <- as.Date(
-      max(unlist(linelist[, date_cols]), na.rm = TRUE),
-      origin = "1970-01-01"
+  stopifnot(
+    "`truncation_time` must be a single positive numeric or a <Date> object." =
```

Comment:
```suggestion
    "`truncation_time` must be a single nonnegative numeric or a <Date> object." =
```

Reply:
Resolved in 199abe8b3db7279116f37dcc498528000a520938. I couldn't commit from the suggestion due to a previous change to the same line.